types/d3-array/index.d.ts

```diff
@@ -196,6 +206,18 @@ export function median<T>(
     accessor: (element: T, i: number, array: Iterable<T>) => number | undefined | null,
 ): number | undefined;
 
+/**
+ * Like median, but returns the index of the element to the left of the median.
+ */
+export function medianIndex(iterable: Iterable<Numeric | undefined | null>): number | undefined;
```

Comment:
```suggestion
export function medianIndex(iterable: Iterable<Numeric | undefined | null>): number;
```

these functions return -1 when there is no defined value in the iterable.